### ISSUE TITLE
Support 'core_version_requirement' for checking compatibility in Drush 8

### ISSUE THREAD

**Issue by smaz87** [drush-ops/drush#4219]
**Is your feature request related to a problem? Please describe.**
Core now has a 'core_version_requirement' key in .info.yml files: https://www.drupal.org/node/3070687

If a module (or theme) requires Drupal 8.7.7 or above, it is not supposed to have the `core: 8x` key anymore. However by not having that, the module can't be enabled via drush due to `drush_extension_check_incompatibility()`

```
function drush_extension_check_incompatibility($file) {
  if (!isset($file->info['core']) || $file->info['core'] != drush_get_drupal_core_compatibility()) {
    return 'Drupal';
  }
  if (version_compare(phpversion(), $file->info['php']) < 0) {
    return 'PHP';
  }
  return FALSE;
}
```
This will fail if the 'core' key isn't set.

If a module tries to have both the `core` and `core_version_requirement` keys set when requiring 8.7.7 or above, the module will fail to install due to Core:
Drupal\Core\Extension\InfoParserException: The 'core_version_requirement' constraint (^8.7.7 || ^9) requires the 'core' key not be set in module/path/module.info.yml in /var/www/html/web/core/lib/Drupal/Core/Extension/InfoParserDynamic.php:55                                                                                                                                         [error]

The module I'm experiencing this issue with is stage_file_proxy 8.x-1.0-rc2: https://www.drupal.org/project/stage_file_proxy/issues/3086276

**Describe the solution you'd like**
`drush_extension_check_incompatibility()` should take into account the new `core_version_requirement` key.

**Describe alternatives you've considered**
Attempted to have both `core:` and `core_version_requirement:` keys as a workaround, but this isn't allowed by core.

**Additional context**
The module enables fine via the UI, or when enabled as part of a `drush cim` - it's only when trying to enable the module directly with drush en`

**Comment by greggles** [comment]
Thanks for posting this and for all the research you've done so far. It's fun using newly released features of core.

**Comment by weitzman** [comment]
I'm inclined to remove the core version check quoted above. Drush sites use Composer and Composer already handles this.

**Comment by weitzman** [comment]
Are folks with this issue using Drush 8? Thats the last version that has [drush_extension_check_incompatibility](http://api.drush.org/api/drush/includes%21environment.inc/function/drush_extension_check_incompatibility/8.0.x). 

I'm not personally inclined to work on Drush 8 issues, but others are welcome to submit PRs and they will be reviewed.

**Comment by greggles** [comment]
Oh, interesting. I did not intentionally test this on drush8 but that is what I see I was using.

```

**Comment by smaz87** [comment]
I'm sure I downloaded & searched the drush10 codebase for it to check if it had been addressed before raising an issue, but saw the function still there... I must have searched in the wrong app / codebase / something. Very sorry!

Looks like I am using drush8 when experiencing the issue.

In the compatibility table, it says Drush8 is supported (though not recommended) for 8.4+. Should 8.7+ be added here, and it change to 'compatible but not supported'? As this issue will prevail for 8.8 (and potentially more so, as more modules switch to supporting D9 in advance)
http://docs.drush.org/en/master/install/#drupal-compatibility

I'm happy try to work on the issue if drush8 is still supported / it's the right thing to do, but if it's better to have people update drush (also in preparation for D9!) then happy to have this closed!
